From: Adam Conrad <adconrad@ubuntu.com>
Date: Wed, 25 May 2022 18:22:00 +0800
Subject: Upstream changes introduced in version 3.0~a24-1ubuntu13

  * Mimic livecd-rootfs's handling of inodes on ext* filesystems to
    reduce the pain of resizing preinstalled images. (LP: #836810)

  * This patch is a dirty hack, and needs some TLC and generalisation
    before it's reasonably upstreamable.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/836810
Last-Update: 2012-08-20
---
 scripts/build/lb_binary_rootfs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/build/lb_binary_rootfs b/scripts/build/lb_binary_rootfs
index 860b600..69e375b 100755
--- a/scripts/build/lb_binary_rootfs
+++ b/scripts/build/lb_binary_rootfs
@@ -129,7 +129,7 @@ case "${LB_CHROOT_FILESYSTEM}" in
 					FAKE_MTAB="true"
 				fi
 
-				Chroot chroot "mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -N ${REAL_INODES} -m 0 filesystem.${LB_CHROOT_FILESYSTEM}"
+				Chroot chroot "mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -i 8192 -m 0 filesystem.${LB_CHROOT_FILESYSTEM}"
 
 				mkdir -p filesystem.tmp
 				${LB_ROOT_COMMAND} mount -o loop chroot/filesystem.${LB_CHROOT_FILESYSTEM} filesystem.tmp
@@ -199,7 +199,7 @@ case "${LB_CHROOT_FILESYSTEM}" in
 
 			false)
 				dd if=/dev/zero of=binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM} bs=1024k count=0 seek=${REAL_DIM}
-				mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -N ${REAL_INODES} -m 0 binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM}
+				mkfs.${LB_CHROOT_FILESYSTEM} -F -b 1024 -i 8192 -m 0 binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM}
 
 				mkdir -p filesystem.tmp
 				${LB_ROOT_COMMAND} mount -o loop binary/${INITFS}/filesystem.${LB_CHROOT_FILESYSTEM} filesystem.tmp
